packages/next/build/swc/src/styled_jsx/mod.rs

```diff
@@ -0,0 +1,786 @@
+        hasher.write(value.as_ref().as_bytes());
+        css = value.to_string().clone();
+        css_span = span.clone();
+        is_dynamic = false;
+      }
+      Expr::Tpl(Tpl {
+        exprs,
+        quasis,
+        span,
+      }) => {
+        if exprs.len() == 0 {
+          hasher.write(quasis[0].raw.value.as_bytes());
+          css = quasis[0].raw.value.to_string();
+          css_span = span.clone();
+          is_dynamic = false;
+        } else {
+          expr.clone().hash(&mut hasher);
+          let mut s = String::new();
+          for i in 0..quasis.len() {
+            let placeholder = if i == quasis.len() - 1 {
+              String::new()
+            } else {
+              format!("__styled-jsx-placeholder__{}", i)
+            };
+            s = format!("{}{}{}", s, quasis[i].raw.value, placeholder)
+          }
+          css = String::from(s);
+          css_span = span.clone();
+          let res = self.evaluator.as_mut().unwrap().eval(&expr);
+          is_dynamic = if let Some(EvalResult::Lit(_)) = res {
+            false
+          } else {
+            true
+          };
+          expressions = exprs.clone();
+        }
+      }
+      Expr::Ident(ident) => {
+        return JSXStyle::External(ExternalStyle {
+          expr: Expr::Member(MemberExpr {
+            obj: ExprOrSuper::Expr(Box::new(Expr::Ident(ident.clone()))),
+            prop: Box::new(Expr::Ident(Ident {
+              sym: "__hash".into(),
+              span: DUMMY_SP,
+              optional: false,
+            })),
+            computed: false,
+            span: DUMMY_SP,
+          }),
+          identifier: ident.clone(),
+          is_global: is_global_jsx_element,
+        });
+      }
+      _ => panic!("Not implemented"), // TODO: handle bad style input
+    }
+
+    return JSXStyle::Local(LocalStyle {
+      hash: format!("{:x}", hasher.finish()),
+      css,
+      css_span,
+      is_dynamic,
+      expressions,
+    });
+  }
+
+  fn replace_jsx_style(&mut self, el: &JSXElement) -> Result<JSXElement, Error> {
+    let style_info = self.styles.pop().unwrap();
+
+    let is_global = el.opening.attrs.iter().any(|attr| {
+      if let JSXAttrOrSpread::JSXAttr(JSXAttr {
+        name: JSXAttrName::Ident(Ident { sym, .. }),
+        ..
+      }) = &attr
+      {
+        if sym == "global" {
+          return true;
+        }
+      }
+      false
+    });
+
+    match &style_info {
+      JSXStyle::Local(style_info) => {
+        let css = transform_css(&style_info, is_global, &self.static_class_name)?;
+        Ok(make_local_styled_jsx_el(
+          &style_info,
+          css,
+          self.style_import_name.as_ref().unwrap(),
+        ))
+      }
+      JSXStyle::External(style) => Ok(make_external_styled_jsx_el(
+        style,
+        self.style_import_name.as_ref().unwrap(),
+      )),
+    }
+  }
+
+  fn get_add_hash(&mut self) -> Option<(String, String)> {
+    let add_hash = self.add_hash.clone();
```

Comment:
You can use `self.add_hash.take()` and it's identical to 

```rust
std::mem::take(&mut self.add_hash)
```

which is

```rust
std::mem::replace(&mut self.add_hash, None)
```

and it's actually taking the value of `self.add_hash` and setting `self.add_hash` to None.